Exception Alert for Social.Messaging.MessageRecipient

## Description of bug
Since upgrade to DNN 9.5.0 I've noticed a new error in the admin log. 

## Steps to reproduce
Not clear at the moment. I think it's when a user archives messages in Message Center.

## Screenshots
![screenshot_1189](https://user-images.githubusercontent.com/27960723/76550426-60f3b600-644f-11ea-9d2c-8f47ea42a9be.png)
![screenshot_1188](https://user-images.githubusercontent.com/27960723/76550450-681ac400-644f-11ea-9a60-8932055e0ea8.png)


## Error information
From Log File:
2020-03-11 00:29:32,742 [generic262][Thread:57][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IndexOutOfRangeException: Archived
   at System.Data.ProviderBase.FieldNameLookup.GetOrdinal(String fieldName)
   at System.Data.SqlClient.SqlDataReader.GetOrdinal(String name)
   at System.Data.SqlClient.SqlDataReader.get_Item(String name)
   at DotNetNuke.Services.Social.Messaging.MessageRecipient.Fill(IDataReader dr)
   at DotNetNuke.Common.Utilities.CBO.FillObjectFromReader(Object objObject, IDataReader dr)

2020-03-11 10:15:49,658 [generic262][Thread:12][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IndexOutOfRangeException: Archived
   at System.Data.ProviderBase.FieldNameLookup.GetOrdinal(String fieldName)
   at System.Data.SqlClient.SqlDataReader.GetOrdinal(String name)
   at System.Data.SqlClient.SqlDataReader.get_Item(String name)
   at DotNetNuke.Services.Social.Messaging.MessageRecipient.Fill(IDataReader dr)
   at DotNetNuke.Common.Utilities.CBO.FillObjectFromReader(Object objObject, IDataReader dr)
## Affected version
9.5.0 is all I have to test with at the moment. 
* [ ] 10.0.0 alpha build
* [ ]  9.5.1 alpha build
* [X]  9.5.0 latest supported release
* [ ]  9.4.4 

## Affected browser
N/A
* [ ] Chrome
* [ ] Firefox
* [ ] Safari
* [ ] Internet Explorer 11
* [ ] Microsoft Edge (Classic)
* [ ] Microsoft Edge Chromium

A user was messaging me earlier today. Went in a looked at the logs and I don’t think this has anything to do with actually archiving of messages. The reason being I have these message now in the log that seem to correlate to myself and this user. It seems like it might be happening during creating of the messaging interface/view. The reason for this thought is looking in the logs and can see my login and then shortly after the error event. Which would correlate to logging and immediately click the message notification in the menu bar. 

2021-02-21 10:27:26,135 [generic262][Thread:52][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IndexOutOfRangeException: Archived
   at System.Data.ProviderBase.FieldNameLookup.GetOrdinal(String fieldName)
   at System.Data.SqlClient.SqlDataReader.GetOrdinal(String name)
   at System.Data.SqlClient.SqlDataReader.get_Item(String name)
   at DotNetNuke.Services.Social.Messaging.MessageRecipient.Fill(IDataReader dr)
   at DotNetNuke.Common.Utilities.CBO.FillObjectFromReader(Object objObject, IDataReader dr)
2021-02-21 10:56:50,444 [generic262][Thread:97][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IndexOutOfRangeException: Archived
   at System.Data.ProviderBase.FieldNameLookup.GetOrdinal(String fieldName)
   at System.Data.SqlClient.SqlDataReader.GetOrdinal(String name)
   at System.Data.SqlClient.SqlDataReader.get_Item(String name)
   at DotNetNuke.Services.Social.Messaging.MessageRecipient.Fill(IDataReader dr)
   at DotNetNuke.Common.Utilities.CBO.FillObjectFromReader(Object objObject, IDataReader dr)
2021-02-21 11:34:41,817 [generic262][Thread:11][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IndexOutOfRangeException: Archived
   at System.Data.ProviderBase.FieldNameLookup.GetOrdinal(String fieldName)
   at System.Data.SqlClient.SqlDataReader.GetOrdinal(String name)
   at System.Data.SqlClient.SqlDataReader.get_Item(String name)
   at DotNetNuke.Services.Social.Messaging.MessageRecipient.Fill(IDataReader dr)
   at DotNetNuke.Common.Utilities.CBO.FillObjectFromReader(Object objObject, IDataReader dr)
2021-02-21 11:40:59,865 [generic262][Thread:65][ERROR] DotNetNuke.Services.Exceptions.Exceptions - System.IndexOutOfRangeException: Archived
   at System.Data.ProviderBase.FieldNameLookup.GetOrdinal(String fieldName)
   at System.Data.SqlClient.SqlDataReader.GetOrdinal(String name)
   at System.Data.SqlClient.SqlDataReader.get_Item(String name)
   at DotNetNuke.Services.Social.Messaging.MessageRecipient.Fill(IDataReader dr)
   at DotNetNuke.Common.Utilities.CBO.FillObjectFromReader(Object objObject, IDataReader dr)

--- a/DNN Platform/Website/Providers/DataProviders/SqlDataProvider/09.10.01.SqlDataProvider
+++ b/DNN Platform/Website/Providers/DataProviders/SqlDataProvider/09.10.01.SqlDataProvider
@@ -1,0 +1,144 @@
+/************************************************************/
+/*****              SqlDataProvider                     *****/
+/*****                                                  *****/
+/*****                                                  *****/
+/***** Note: To manually execute this script you must   *****/
+/*****       perform a search and replace operation     *****/
+/*****       for {databaseOwner} and {objectQualifier}  *****/
+/*****                                                  *****/
+/************************************************************/
+
+
+IF EXISTS (SELECT * FROM sys.Procedures WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}CoreMessaging_GetNextMessagesForInstantDispatch]'))
+    DROP PROCEDURE {databaseOwner}[{objectQualifier}CoreMessaging_GetNextMessagesForInstantDispatch];
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}CoreMessaging_GetNextMessagesForInstantDispatch]
+    @SchedulerInstance UNIQUEIDENTIFIER,
+    @BatchSize         INT
+AS 
+BEGIN
+    -- reset possibly remaining records from any previous run of this SchedulerInstance:
+    UPDATE {databaseOwner}[{objectQualifier}CoreMessaging_MessageRecipients]
+       SET [EmailSchedulerInstance] = Null,
+           [LastModifiedOnDate]     = GetDate()
+     WHERE [EmailSchedulerInstance] = @SchedulerInstance
+       AND [EmailSent] = 0 AND [Read] = 0 AND [Archived] = 0;
+       
+    -- reset possibly remaining outdated records from other instances:
+    UPDATE {databaseOwner}[{objectQualifier}CoreMessaging_MessageRecipients]
+     SET   [EmailSchedulerInstance] = Null
+     WHERE [EmailSent] = 0 AND [Read] = 0 AND [Archived] = 0
+       AND [EmailSchedulerInstance] Is Not Null AND [LastModifiedOnDate] < DateAdd(hh, -2, GetDate());
+       
+
+    -- mark messages for dispatch, so they won't be handled by another SchedulerInstance:
+    UPDATE TOP (@BatchSize) R
+     SET   [EmailSchedulerInstance] = @SchedulerInstance,
+           [LastModifiedOnDate]     = GetDate()
+     FROM       {databaseOwner}[{objectQualifier}CoreMessaging_MessageRecipients] AS R 
+     INNER JOIN {databaseOwner}[{objectQualifier}CoreMessaging_Messages]          AS M ON R.MessageID = M.MessageID
+     LEFT  JOIN {databaseOwner}[{objectQualifier}CoreMessaging_UserPreferences]   AS P ON R.UserID    = P.UserID    AND M.PortalID = P.PortalID
+     WHERE R.[EmailSent] = 0 AND R.[Read] = 0 AND R.[Archived] = 0 AND EmailSchedulerInstance IS NULL
+       AND CASE 
+            WHEN M.NotificationTypeID IS Null 
+            THEN IsNull(P.[MessagesEmailFrequency],      0) -- direct mails are sent immediately by default
+            ELSE IsNull(p.[NotificationsEmailFrequency], 2) -- notifications are sent as daily digest by default
+           END = 0;
+
+    SELECT M.[PortalID],
+           M.[NotificationTypeID],
+           M.[To],
+           M.[From],
+           M.[Subject],
+           M.[Body],
+           M.[SenderUserID],
+           M.[ExpirationDate],
+           M.[Context],
+           R.[RecipientID],
+           R.[MessageID],
+           R.[EmailSent],
+           R.[Read],
+           R.[Archived],
+           R.[EmailSchedulerInstance],
+           R.[UserID],
+		   R.[CreatedByUserID],
+		   R.[CreatedOnDate],
+		   R.[LastModifiedByUserID],
+		   R.[LastModifiedOnDate],
+		   R.[SendToast]
+     FROM  {databaseOwner}[{objectQualifier}CoreMessaging_MessageRecipients] R
+     JOIN  {databaseOwner}[{objectQualifier}CoreMessaging_Messages]          M ON R.MessageID = M.MessageID
+     WHERE [EmailSent] = 0 -- Filter these columms 4 to use proper index
+       AND [Read]      = 0
+       AND [Archived]  = 0
+       AND [EmailSchedulerInstance] = @SchedulerInstance
+     ORDER BY --[PortalID],
+              [UserID],
+              [RecipientID]
+END; -- Procedure
+GO
+
+
+IF EXISTS (SELECT * FROM sys.Procedures WHERE object_id = OBJECT_ID(N'{databaseOwner}[{objectQualifier}CoreMessaging_GetNextMessagesForDigestDispatch]'))
+    DROP PROCEDURE {databaseOwner}[{objectQualifier}CoreMessaging_GetNextMessagesForDigestDispatch];
+GO
+
+IF EXISTS (SELECT * FROM sys.Procedures WHERE object_id = OBJECT_ID(N'[dbo].[CoreMessaging_GetNextMessagesForDigestDispatch]'))
+    DROP PROCEDURE [dbo].[CoreMessaging_GetNextMessagesForDigestDispatch];
+GO
+
+CREATE PROCEDURE {databaseOwner}[{objectQualifier}CoreMessaging_GetNextMessagesForDigestDispatch]
+    @Frequency         INT,
+    @SchedulerInstance UNIQUEIDENTIFIER,
+    @BatchSize         INT
+AS 
+BEGIN
+    UPDATE R
+     SET   [EmailSchedulerInstance] = @SchedulerInstance,
+           [LastModifiedOnDate]     = GetDate()
+     FROM  {databaseOwner}[{objectQualifier}CoreMessaging_MessageRecipients] R
+     JOIN  (SELECT TOP (@BatchSize)
+                   UserID
+             FROM  {databaseOwner}[{objectQualifier}vw_MessagesForDispatch] 
+             WHERE [EmailSchedulerInstance] IS NULL
+             AND   [EmailFrequency] = @Frequency
+             GROUP BY UserID
+             ORDER BY UserID) D ON R.UserID = D.UserID 
+
+    SELECT M.[PortalID],
+           M.[NotificationTypeID],
+           M.[To],
+           M.[From],
+           M.[Subject],
+           M.[Body],
+           M.[SenderUserID],
+           M.[ExpirationDate],
+           M.[Context],
+           R.[RecipientID],
+           R.[MessageID],
+           R.[UserID],
+           R.[EmailSent],
+           R.[Read],
+           R.[Archived],
+           R.[EmailSchedulerInstance],
+		   R.[CreatedByUserID],
+		   R.[CreatedOnDate],
+		   R.[LastModifiedByUserID],
+		   R.[LastModifiedOnDate],
+		   R.[SendToast]
+     FROM  {databaseOwner}[{objectQualifier}CoreMessaging_MessageRecipients] R
+     JOIN  {databaseOwner}[{objectQualifier}CoreMessaging_Messages]          M ON R.MessageID = M.MessageID
+     WHERE [EmailSent] = 0 -- Filter these 4 columms to use proper index
+       AND [Read]      = 0
+       AND [Archived]  = 0
+       AND [EmailSchedulerInstance] = @SchedulerInstance
+     ORDER BY --[PortalID],
+              [UserID],
+              [RecipientID] DESC
+END; -- Procedure
+GO
+
+/************************************************************/
+/*****              SqlDataProvider                     *****/
+/************************************************************/

